llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -14147,9 +14149,14 @@ void SITargetLowering::AdjustInstrPostInstrSelection(MachineInstr &MI,
     // use between vgpr and agpr as agpr tuples tend to be big.
     if (!MI.getDesc().operands().empty()) {
       unsigned Opc = MI.getOpcode();
+      bool NoAGPRs = !Info->mayNeedAGPRs();
+      SmallVector<int> Opnds;
```

Comment:
Can you avoid using SmallVector here? I'd add src2 to the static array and break from the loop instead. This is cheaper. Actually I'd probably add isMAI() check too to the whole if (in a separate NFCI).

Reply:
> Can you avoid using SmallVector here? I'd add src2 to the static array and break from the loop instead. This is cheaper. Actually I'd probably add isMAI() check too to the whole if (in a separate NFCI).

Done